test/GPv2UniswapRouter.test.ts

```diff
@@ -76,11 +107,259 @@ describe("GPv2UniswapRouter", () => {
     );
   }
 
+  async function deployMockPair(
+    tokenA: string,
+    tokenB: string,
+  ): Promise<MockContract> {
+    const { abi } = await artifacts.readArtifact("IUniswapV2Pair");
+    const pair = await waffle.deployMockContract(deployer, abi);
+    await uniswapFactory.mock.getPair
```

Comment:
An alternative way to do this would be to use impersonated accounts to make sure the counterfactual address is the same, but this seems more flexible 👍

Reply:
I couldn't figure out how to set code for an impersonated address (which is needed to handle the `getReserves` call), but otherwise, I agree impersonating would be better.